### PR TITLE
swarm: log unexpected listener errors

### LINKED ISSUE #2264: Log error in accept
https://github.com/libp2p/go-libp2p/blob/8313917571fec7d26813ff437013fde6f48659f7/p2p/net/swarm/swarm_listen.go#L130

ignore swarm close errors.

### PR DESCRIPTION
**Pull request by MarcoPolo**
Closes #2264

### CODE REVIEW

File: p2p/transport/quicreuse/listener.go [hunk 1]
```diff
@@ -134,6 +136,9 @@ func (l *connListener) Run() error {
 	for {
 		conn, err := l.l.Accept(context.Background())
 		if err != nil {
+			if err == quic.ErrServerClosed || strings.Contains(err.Error(), "use of closed network connection") {
```
Comment (marten-seemann) [inline p2p/transport/quicreuse/listener.go:139]: `errors.Is`?

File: p2p/net/swarm/swarm_listen.go [hunk 2]
```diff
@@ -127,6 +127,9 @@ func (s *Swarm) AddListenAddr(a ma.Multiaddr) error {
 		for {
 			c, err := list.Accept()
 			if err != nil {
+				if err != transport.ErrListenerClosed {
+					log.Error("swarm listener accept error: ", err)
```
Comment (marten-seemann) [inline p2p/net/swarm/swarm_listen.go:131]: Should this include which listener this is? For example:
```suggestion
					log.Errorf("swarm listener for %s accept error: %s", a, err)
```

**Review by marten-seemann — APPROVED** [review]
A couple of nits, otherwise this lgtm.

File: p2p/net/swarm/swarm_listen.go (hunk 2 above)
Comment (MarcoPolo) [reply p2p/net/swarm/swarm_listen.go:131]: You're right. I misread this as `log.Errorf("swarm listener %s accept error: %s", list, err)`. That makes sense.

**Review by marten-seemann — APPROVED** [review]

**Pull request merged**